app/models/users/setting.rb

```diff
@@ -0,0 +1,56 @@
+module Users
+  class Setting < ApplicationRecord
+    self.table_name_prefix = "users_"
+
+    EDITORS = %w[v1 v2].freeze
```

Comment:
If we made these constants `enum`s we'd automatically get things like scopes which may come in handy:  

```ruby
enum editor_version: { v1: 0, v2: 1 }
```

You'd still be able to use them in the validations, e.g.  `inclusion: { in: editor_versions.keys }`.

Reply:
I'm on it 👍🏾

Reply:
@citizen428 @Ridhwana

Question:
In `app/models/users/setting.rb`, I replaced the constants `EDITORS`, `FONTS`, `INBOXES`, `NAVBARS` & `THEMES` with enums as below:

```ruby
enum editor_versions: { v1: 0, v2: 1 }
enum config_fonts: { comic_sans: 0, default: 1, monospace: 2, open_dyslexic: 3, sans_serif: 4, serif: 5 }
enum config_inbox_type: { open: 0, private: 1 }
enum config_navbars: { default: 0, static: 1 }
enum config_themes: { default: 0, minimal_light_theme: 1, night_theme: 2, pink_theme: 3, ten_x_hacker_theme: 4 }
```

Now, in the migration file `xxx_create_users_settings.rb`, I thank that I need to add the columns that will hold these enum values.
But I'm not sure if I simply add a `t` row for each enum like:
```ruby
t.integer "editor_version", default: 1, null: false
```

Or do I need to replace the existing string associated with each setting?

Meaning, for the `editor_version` setting, do I delete
```ruby
t.string "editor_version", default: "v1"
```
and replace it with
```ruby
t.integer "editor_version", default: 1, null: false
```

I'm reading up on how enums work, but I thought to ask here as well 🙏🏾

Reply:
@msarit In short, yes, you replace them.

1. If you already ran the migration, you have to roll it back, e.g. with `rails db:rollback`.
2. You add integer columns for each `enum` (the default should probably be 0, not 1), e.g.
    ```ruby
    t.integer "editor_version", default: 0, null: false
    ```
3. You can remove the string columns, they are no longer needed for anything.

Reply:
Just a small update on step 1 above (you may or may not know this already but thought I'd outline it here anyways). In order to rollback the migration, you will most likely need to run `rake db:rollback STEP=n` where n is the number of migrations you need to rollback. So in this PR (if you haven't merged with master as yet) then 'create_user_settings' is the second last migration, so you'll need to run `rake db:rollback STEP=2` to get to a state where you can change the user_settings migration file.

Reply:
You can also just rollback a specific version, which only works if two migrations aren't "additive", as in they don't depend on each other. `VERSION=XYZ bundle exec rails db:migrate:down`, to find all migration versions type: `bundle exec rails db:migrate:status`